test/Microsoft.Azure.ServiceBus.UnitTests/BrokeredMessageTests.cs

```diff
@@ -105,5 +106,82 @@ void CustomMessageIdGenerator()
+            }
+
+            [Theory]
+            [DisplayTestMethodName]
+            [InlineData(ReceiveMode.ReceiveAndDelete)]
+            [InlineData(ReceiveMode.PeekLock)]
+            async Task Should_return_true_for_message_that_was_sent_and_received(ReceiveMode receiveMode)
+            {
+                var queueClient = QueueClient.CreateFromConnectionString(TestUtility.GetEntityConnectionString(Constants.NonPartitionedQueueName), receiveMode);
+                try
+                {
+                    await TestUtility.SendMessagesAsync(queueClient.InnerSender, 1);
+                    var receivedMessages = await TestUtility.ReceiveMessagesAsync(queueClient.InnerReceiver, 1);
+                    Assert.True(receivedMessages.First().IsReceived);
+
+                    // TODO: remove when per test cleanup is possible
+                    if (receiveMode == ReceiveMode.PeekLock)
+                    {
+                        await receivedMessages.First().CompleteAsync();
+                    }
+                }
+                finally
+                {
+                    await queueClient.CloseAsync();
+                }
+            }
+
+            [Fact]
+            [DisplayTestMethodName]
+            async Task Should_return_true_for_peeked_message()
+            {
+                var queueClient = QueueClient.CreateFromConnectionString(TestUtility.GetEntityConnectionString(Constants.NonPartitionedQueueName), ReceiveMode.PeekLock);
+                try
+                {
+                    await TestUtility.SendMessagesAsync(queueClient.InnerSender, 1);
+                    var peekedMessage = await TestUtility.PeekMessageAsync(queueClient.InnerReceiver);
+                    var result = peekedMessage.IsReceived;
+                    Assert.True(result);
+                }
+                finally
+                {
+                    var messages = await TestUtility.ReceiveMessagesAsync(queueClient.InnerReceiver, 1);
+                    await TestUtility.CompleteMessagesAsync(queueClient.InnerReceiver, messages);
+                    await queueClient.CloseAsync();
+                }
+            }
+
+            [Fact]
+            [DisplayTestMethodName]
+            async Task Should_return_true_for_completed_message()
+            {
+                var queueClient = QueueClient.CreateFromConnectionString(TestUtility.GetEntityConnectionString(Constants.NonPartitionedQueueName), ReceiveMode.PeekLock);
+                try
+                {
+                    await TestUtility.SendMessagesAsync(queueClient.InnerSender, 1);
+                    var peekedMessage = await TestUtility.PeekMessageAsync(queueClient.InnerReceiver);
+                    var messages = await TestUtility.ReceiveMessagesAsync(queueClient.InnerReceiver, 1);
+                    await TestUtility.CompleteMessagesAsync(queueClient.InnerReceiver, messages);
+                    var result = peekedMessage.IsReceived;
```

Comment:
@SeanFeldman, the thing is complete call does not really do any modification on the message after it was received.. It just takes lockToken on the message and sends it to Service to complete the message. So there isnt really a validation necessary?